hs/app/reach/Main.hs

```diff
@@ -2465,17 +2465,35 @@ instance FromJSON GitHubGistResponse where
   parseJSON = withObject "GitHubGistResponse" $ \o -> GitHubGistResponse <$> o .: "html_url"
 
 support :: Subcommand
-support = command "support" $ info (pure g) d
+support = command "support" $ info h d
   where
-    d = progDesc "Create GitHub gist of index.rsh and index.mjs"
+    d = progDesc "Create a GitHub gist of index.rsh and index.mjs (default), or specify your own files to upload!"
+    h = go <$> supportFromCommandLineHs
+    go SupportToolArgs {sta_so = SupportOpts {}} = do
+      rawArgs <- liftIO getArgs
+      let rawArgs' = dropWhile (/= "support") rawArgs
+      let actualArgs = tailMay rawArgs'
+      case actualArgs of
+        Nothing -> impossible $ "support args do not start with 'support': " <> show rawArgs
+        Just [] -> doDefaultBehavior
+        Just xs -> liftIO $ do
+          arrayOfPairs <- mapM z xs
+          case [] `elem` arrayOfPairs of
+            False -> upload arrayOfPairs
+            True -> do
+              putStrLn "\nNothing uploaded"
+              exitWith $ ExitFailure 1
     f i c = i .= object
       [ "content" .= if T.null (T.strip $ pack c) then "// (Empty source file)" else c
       , "language" .= ("JavaScript" :: String)
       , "type" .= ("application/javascript" :: String)
       ]
     z i = doesFileExist i >>= \case
-      False -> pure []
-      True -> (\a -> [f (K.fromString i) a]) <$> readFile i
+      False -> do
+          putStrLn $ "Couldn't find the following file: " ++ i
+          pure []
```

Comment:
You should actually error, rather than just print out a message

Reply:
I wanted to error after doing `z` on every file, as opposed to during, so users could see all erroneous file paths at once and fix them all in one go as opposed to repeatedly running `reach support` and seeing and fixing faulty file paths one at a time.

[340b27f](https://github.com/reach-sh/reach-lang/pull/1063/commits/340b27fa2f57d4bdd55ba413777af360e13f5a6e) does error during `z`.